### ISSUE TITLE
Error when using Ref for Certificate Arns.

### ISSUE THREAD

**Issue by resistdesign** [martysweet/cfn-lint#247]
If you declare a Certificate Arn with a Ref to a Parameter like this:

```yaml
  AppCDNCloudFront:
    Type: AWS::CloudFront::Distribution
    DependsOn: AppHostingStorage
    Properties:
      DistributionConfig:
        Aliases:
          - Ref: AppDomain
        ViewerCertificate:
          AcmCertificateArn:
            # CERT ARN:
            Ref: SSLCertificateArn

# OR...

  APIGatewayDomainName:
    Type: AWS::ApiGateway::DomainName
    Properties:
      CertificateArn:
        # CERT ARN:
        Ref: SSLCertificateArn
```

You get the following critical error:

` Expecting an ARN, got 'string_input_SSLCertificateArn'`

I think the expected result would be that it would accept it as long as `SSLCertificateArn` was a valid parameter id.
